lib/widgets/medication_details.dart

```diff
@@ -0,0 +1,670 @@
+                                value: vt['name'],
+                                child: Text(vt['name'] ?? 'unknown')))
+                        .toList(),
+                    validator: (value) => value == null ? frequencyRequired : null,
+                  );
+                },
+                valueListenable: _typeChangeAllowed,
+              ))
+        ]));
+  }
+
+  Widget _durationUnitRow(List<dynamic> items) {
+    return Container(
+      alignment: Alignment.centerRight,
+      width: 150,
+      child: ValueListenableBuilder<bool>(
+        builder:
+            (BuildContext context, bool allowChange, Widget? child) {
+          return DropdownButtonFormField(
+            value: _selectedDurationType,
+            alignment: Alignment.centerRight,
+            onChanged: (newvalue) {
+              setState(() {
+                _selectedDurationType = newvalue!;
+                _updateResult();
+              });
+            },
+            items: (items.where((element) => durationUnits.contains(element['name'])))
+                .map<DropdownMenuItem<String>>((vt) =>
+                    DropdownMenuItem<String>(
+                        value: vt['name'],
+                        child: Text(vt['name'] ?? 'unknown')))
+                .toList(),
+            validator: (value) => value == null ? durationUnitRequired : null,
+          );
+        },
+        valueListenable: _typeChangeAllowed,
+      ));
+  }
+
+  Padding _dosingInstructionsRow(String lblTitle, List<dynamic> items) {
+    return Padding(
+        padding: const EdgeInsets.symmetric(horizontal: 16.0),
+        child:
+            Row(mainAxisAlignment: MainAxisAlignment.spaceBetween, children: [
+          Text(lblTitle),
+          Container(
+              alignment: Alignment.centerRight,
+              width: 200,
+              child: ValueListenableBuilder<bool>(
+                builder:
+                    (BuildContext context, bool allowChange, Widget? child) {
+                  return DropdownButtonFormField<String>(
+                    value: _selectedDosingInstructions,
+                    alignment: Alignment.centerRight,
+                    onChanged: (newvalue) {
+                      setState(() {
+                        _selectedDosingInstructions = newvalue!;
+                      });
+                    },
+                    items: (items)
+                        .map<DropdownMenuItem<String>>((vt) =>
+                            DropdownMenuItem<String>(
+                                value: vt['name'],
+                                child: Text(vt['name'] ?? 'unknown')))
+                        .toList(),
+                    validator: (value) => value == null ? dosingInstructionRequired : null,
+                  );
+                },
+                valueListenable: _typeChangeAllowed,
+              ))
+        ]));
+  }
+
+  Padding _numericRow(String lblTitle, TextEditingController controller) {
```

Comment:
better name? _numericRow() does not express what it is rendering .. if required break up into multiple methods but each expressing what they are rendering

Reply:
Changed